csrc/kernel.cpp

```diff
@@ -314,6 +314,13 @@ void Kernel::finalize(std::vector<Expr*> top_level_exprs) {
   for (auto alloc : summary_.global_allocations) {
     parameters_.push_back(alloc->buffer());
   }
+  // If any cluster parallel type is used set has_thread_block_cluster
+  for (auto pt : kParallelTypeCIDs) {
+    if (GpuLower::current()->parallelDimensionMap().get(pt)) {
+      summary_.has_thread_block_cluster = true;
```

Comment:
I'm worried about the case where there is a cluster that end-up having a dimension {1,1,1} in a kernel. In this case we don't want to launch a CGA kernel because this will introduce some limitations. In that case (clusterDim = {1,1,1}) it's better to drop cluster and launch a regular kernel